src/internal.c

```diff
@@ -23198,9 +23205,89 @@ int SetCipherList(WOLFSSL_CTX* ctx, Suites* suites, const char* list)
             }
         }
 
+        if (length > 1) {
+            if (*current == '!') {
+                allowing = 0;
+                current++;
+                length--;
+            }
+        }
+
         XSTRNCPY(name, current, length);
         name[(length == sizeof(name)) ? length - 1 : length] = 0;
 
+        if (XSTRCMP(name, "DEFAULT") == 0) {
```

Comment:
Should this new code be wrapped with `OPENSSL_EXTRA` as to not grow code size?

Reply:
Good idea. Especially with all the strings. I'll try to refactor the suite selection a bit as well.